pkg/types/jsonpatch.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
```

Comment:
This package was never meant to be a home for all new types - it's just a failure on my part to make a better name.

I suggest you start a new package `pkg/patching` or `pkg/patch` and define the new types there, and let this package import from there.  That allows more freedom in naming structs and fields.

Eventually, the transformer code for patching can go there too.  There's no good reason to keep all transformer code in one place.

Reply:
sounds good